File: MPP/Lab3_2/Network/src/main/java/Proxy/RezervareProxy.java
package Proxy;

import ProjectModels.Rezervare;
import ServiceInterface.IServiceRezervare;
import Utils.Request;
import Utils.Response;

import java.io.IOException;
import java.io.ObjectInputStream;
import java.io.ObjectOutputStream;
import java.net.Socket;
import java.util.List;

public class RezervareProxy implements IServiceRezervare {
    private ObjectInputStream input;
    private ObjectOutputStream output;
    private Socket connection;
    private ReadResponse r;

    public RezervareProxy(ObjectInputStream input, ObjectOutputStream output, Socket connection, ReadResponse r) {
        this.input = input;
        this.output = output;
        this.connection = connection;
        this.r = r;
    }

    @Override
    public List<Rezervare> getList(int idDestinatie) {
        Request cer = new Request("GetListaRezervari");
        cer.setId(idDestinatie);
        try{
            output.writeObject(cer);
            output.flush();
            Response re = r.readResponse();
            if(re.getMesaj().equals("ok"))
            {
                return re.getList();
            }
        } catch (IOException e) {
            e.printStackTrace();
        }
        return null;
    }

    @Override
    public boolean save(Rezervare rez) {
        Request cer = new Request("Save");
        cer.setRezervare(rez);
        try{
            output.writeObject(cer);
            output.flush();
            Response re = r.readResponse();
            if(re.getMesaj().equals("ok"))
            {
                return re.getBol();
            }
        } catch (IOException e) {
            e.printStackTrace();
        }
        return false;
    }
}

File: MPP/ServerC#CrossPlatform/ConsoleApplication1/Networking/RezervareProxy.cs
using System;
using System.Collections.Generic;
using System.Data;
using System.Linq;
using System.Net.Sockets;
using System.Runtime.Serialization.Formatters.Binary;
using System.Text;
using System.Threading.Tasks;
using Model;
using Service;

namespace Networking
{
    public class RezervareProxy:IServiceRezervare
    {
        NetworkStream stream;
        BinaryFormatter formatter;
        TcpClient connection;
        ReadRespone r;

        public  RezervareProxy(NetworkStream s , TcpClient c , ReadRespone rr)
        {
            r = rr;
            stream = s;
            connection = c;
            formatter = new BinaryFormatter();
        }

        public DataSet getRezervazri( int id)
        {
            Request req = new Request("getRezervazri");
            req.setId(id);
            formatter.Serialize(stream, req);
            stream.Flush();
            Response re = r.readResponse();
            if (re.getMesaj().Equals("ok"))
            {
                return re.getDs();
                
            }
            return null;
        }

        public bool save(Rezervare rez)
        {
            Request req = new Request("Save");
            req.setRezervare(rez);
            formatter.Serialize(stream, req);
            stream.Flush();
            Response re = r.readResponse();
            if(re.getMesaj().Equals("ok"))
            {
                return re.getBol();
            }
            return false;  
        }

        
    }
}
